lib/src/content/wolt_modal_sheet_animated_switcher.dart

```diff
@@ -96,6 +136,42 @@ class _WoltModalSheetAnimatedSwitcherState
     ];
   }
 
+  void _resetScrollControllers() {
+    _scrollControllers.clear();
+    _scrollControllers = [
+      for (int i = 0; i < _pagesCount; i++)
+        (_page.scrollController ??
+            ScrollController(initialScrollOffset: _scrollPositions[i].value))
+    ];
+  }
+
+  void _subscribeToCurrentPageScrollPositionChanges() {
+    for (final scrollController in _scrollControllers) {
+      scrollController.addListener(() {
+        if (_currentPageScrollController.hasClients) {
+          _currentPageScrollPosition.value =
+              _currentPageScrollController.position.pixels;
+        }
+      });
+    }
+  }
+
+  void _subscribeToSoftKeyboardClosedEvent() {
+    _softKeyboardVisibilitySubscription =
+        KeyboardVisibilityController().onChange.listen((
+      bool visible,
+    ) async {
+      if (!visible) {
+        /// Wait for closing soft keyboard animation to finish before emitting new value.
+        await Future.delayed(const Duration(milliseconds: 250));
```

Comment:
Is it possible that different devices will have different keyboard open/close duration which might affect this? I wonder is it possible to read information this from system/device

Reply:
i've the same concern. This needs to be more testing to be sure.

Reply:
@TahaTesser how is this problem approached in Flutter in general? I will have a second look to address this.

Reply:
Framework gets native callbacks for events that happen on the system level.